src/python_bluesky_taskgraph/core/task_graph.py

```diff
@@ -1,13 +1,29 @@
-from typing import Dict, List, Set
+from dataclasses import dataclass
+from typing import Callable, Dict, List, Set, Union
 
 from python_bluesky_taskgraph.core.task import BlueskyTask
+from python_bluesky_taskgraph.tasks.stub_tasks import CloseRunTask, OpenRunTask
+
+
+@dataclass
+class TaskTuple:
+    task: BlueskyTask
+    inputs: List[str]
```

Comment:
Does this relate to `TypedInput` below?

Reply:
No relation to TypedInput, TypedInput is a utility type to facilitate Tasks with a single argument (and might be left on the cutting room floor before too long). e.g. 

PlantRequiringTask(BlueskyTask[TypedInput[Plant]]): as "short"cut for 

PlantRequiringTask(BlueskyTask['PlantInput']): 
  @dataclass 
  PlantInput: 
    plant: Plan


TaskTuple/PreparedTask just triplet of task, input names, output names. Always List[str].